internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/DaVinciClientMemoryLimitTest.java

```diff
@@ -142,122 +159,168 @@ private VeniceProperties getDaVinciBackendConfig(
-    String storeName = Utils.getUniqueString("davinci_memory_limit_test");
+  /**
+   * TODO: If we deflake the 2 disabled tests in this class, we should make them leverage the stores setup here, or else
+   * move them to another class...
+   */
+  @BeforeMethod
+  public void storeSetup() throws IOException {
+    this.storeName = Utils.getUniqueString("davinci_memory_limit_test");
+    this.storeNameWithoutMemoryEnforcement = Utils.getUniqueString("store_without_memory_enforcement");
+
     // Test a small push
     File inputDir = getTempDataDirectory();
     String inputDirPath = "file://" + inputDir.getAbsolutePath();
     Schema recordSchema = TestWriteUtils.writeSimpleAvroFileWithStringToStringSchema(inputDir, 100, 100);
-    Properties vpjProperties = defaultVPJProps(veniceCluster, inputDirPath, storeName);
 
-    String storeNameWithoutMemoryEnforcement = Utils.getUniqueString("store_without_memory_enforcement");
+    this.vpjProperties = defaultVPJProps(veniceCluster, inputDirPath, this.storeName);
+    setupStore(this.storeName, this.vpjProperties, recordSchema);
 
-    try (
-        ControllerClient controllerClient =
-            createStoreForJob(veniceCluster.getClusterName(), recordSchema, vpjProperties);
-        AvroGenericStoreClient client = ClientFactory.getAndStartGenericAvroClient(
-            ClientConfig.defaultGenericClientConfig(storeName).setVeniceURL(veniceCluster.getRandomRouterURL()))) {
-      veniceCluster.createMetaSystemStore(storeName);
-      veniceCluster.createPushStatusSystemStore(storeName);
+    Properties otherProps = defaultVPJProps(veniceCluster, inputDirPath, this.storeNameWithoutMemoryEnforcement);
+    setupStore(this.storeNameWithoutMemoryEnforcement, otherProps, recordSchema);
 
-      // Make sure DaVinci push status system store is enabled
-      StoreResponse storeResponse = controllerClient.getStore(storeName);
-      assertFalse(storeResponse.isError(), "Store response receives an error: " + storeResponse.getError());
+    this.client = ClientFactory.getAndStartGenericAvroClient(
+        ClientConfig.defaultGenericClientConfig(storeName).setVeniceURL(veniceCluster.getRandomRouterURL()));
+
+    this.factory = null; // will be constructed in the test
+  }
+
+  private void setupStore(String store, Properties vpjProperties, Schema recordSchema) {
+    this.veniceCluster.createStoreForJob(recordSchema, vpjProperties);
+    this.veniceCluster.createMetaSystemStore(store);
+    this.veniceCluster.createPushStatusSystemStore(store);
+
+    // Make sure system stores are enabled
+    this.veniceCluster.useControllerClient(cc -> {
+      StoreResponse storeResponse = TestUtils.assertCommand(cc.getStore(store));
       assertTrue(storeResponse.getStore().isDaVinciPushStatusStoreEnabled());
-      prepareMetaSystemStore(storeName);
-      // Do an VPJ push
-      runVPJ(vpjProperties, 1, controllerClient);
+      assertTrue(storeResponse.getStore().isStoreMetaSystemStoreEnabled());
```

Comment:
Just curious - isn't preparemetaSystemStore call equal to check meta store flag enabled? Or could you explain a bit what's the change that deflake the test?

Reply:
I see, thank you for the explanation!